tree/treeplayer/src/TBranchProxyDirector.cxx

```diff
@@ -136,17 +136,20 @@ namespace Internal {
 
       TTree* oldtree = fTree;
       fTree = newtree;
-      Notify();
+      if(!Notify()) return nullptr;
       return oldtree;
    }
 
    Bool_t TBranchProxyDirector::Notify() {
       fEntry = -1;
-
+      bool retVal = true;
       for_each(fDirected.begin(),fDirected.end(),NotifyDirected);
+      for (auto brProxy : fDirected) {
+         retVal = brProxy->Notify();
```

Comment:
This means that we are returning an error only if the last proxy has an error.

Reply:
:( my bad. Nice catch.